gdc_ng_models/alembic/versions/12dbbcac7a1d_change_columns_to_biginteger.py

```diff
@@ -0,0 +1,52 @@
+"""change columns to BigInteger
+
+Revision ID: 12dbbcac7a1d
+Revises: e9d53a640d5d
+Create Date: 2021-03-30 13:16:01.055456
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "12dbbcac7a1d"
+down_revision = "e9d53a640d5d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.alter_column("notifications", "id", type_=sa.BigInteger)
+    op.alter_column("qc_test_runs", "id", type_=sa.BigInteger)
+    op.alter_column("qc_validation_results", "id", type_=sa.BigInteger)
+    op.alter_column("qc_validation_results", "test_run_id", type_=sa.BigInteger)
+    op.alter_column("redaction_log", "id", type_=sa.BigInteger)
+    op.alter_column("redaction_entry", "redaction_id", type_=sa.BigInteger)
+    op.alter_column("released_data_log", "id", type_=sa.BigInteger)
+    op.alter_column("gdc_reports", "id", type_=sa.BigInteger)
+    op.alter_column("study_rule", "id", type_=sa.BigInteger)
+    op.alter_column("study_rule_program", "study_rule_id", type_=sa.BigInteger)
+    op.alter_column("study_rule_program_project", "study_rule_id", type_=sa.BigInteger)
+    op.alter_column("transaction_logs", "id", type_=sa.BigInteger)
+    op.alter_column("transaction_snapshots", "transaction_id", type_=sa.BigInteger)
+    op.alter_column("transaction_documents", "id", type_=sa.BigInteger)
+    op.alter_column("transaction_documents", "transaction_id", type_=sa.BigInteger)
```

Comment:
Would adding the parameter `existing_type=sa.Integer` help fortify this alteration? I'm not sure if I'm reading [the docs](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.alter_column.params.existing_type) right.

Reply:
yea i had seen that it was mostly if we used MySQL or if the column being changed was a boolean or enum. It might make it clearer for future users to have existing_type set, but I decided for a shorter method call since it was optional/didn't affect the migration (i don't think).

Reply:
I mean in the sense that the alembic migration won't attempt to change the column to a bigint unless it is currently an int. If space is a concern this can fit into a for loop.

```python
tables = ["notifications", "qc_test_runs", "qc_validation_results", ...]
for table in tables:
    op.alter_column(
        table, 
        "id", 
        existing_type=sa.Integer,
        type_=sa.BigInteger,
    )
```

Reply:
Looks like, at least for Postgres, the optional flag `exisiting_type`  is not used at all (even if specified) to check whether the column has type == `existing_type`. Ran an example migration that would turn  convert the `id` column from `BigInteger` back to `Integer`. 

`op.alter_column("notifications", "id", existing_type=sa.Integer, type_=sa.Integer)`

I specified here that the existing_type is `Integer`, which is not accurate (it's `BigInteger`). The migration was successful regardless, with this SQL statement being emitted:

`ALTER TABLE notifications ALTER COLUMN id TYPE INTEGER`

Seems like for Postgres that the `existing_type` flag is not being used.